### ISSUE TITLE
Upgrade BatchNorm.cpp so the stats could be frozen (sharing bn)

### ISSUE THREAD

**Issue by fvarno** [pytorch/pytorch#10856]
Hello developers,
Currently the only way of freezing the batchnorm stats dynamically is to play with training flag. But one may need to stop updating batchnorm's stats at some point and still be able to train delta_gammas and betas. This is addressed in Tensorflow with "reuse" flag. Please consider updating BatchNorm.cpp, nn.Functional and nn.modules.batchnorm.

Regards,
Farshid

**Comment by SsnL** [comment]
This is supported already. In `eval()` mode you can still get gradients. Alternatively, you can set `track_running_stats=False`.
